backend/src/services/auth.rs

```diff
@@ -0,0 +1,170 @@
+use crate::router::ApiV1;
+use crate::router::ApiVersion;
+use crate::{
+    connector::BobClient,
+    models::shared::{BobConnectionData, Hostname, RequestTimeout},
+    prelude::*,
+};
+use axum::{
+    async_trait,
+    extract::State,
+    headers::{authorization::Basic, Authorization},
+    Extension, Json,
+};
+use axum_login::{
+    extractors::AuthContext, memory_store::MemoryStore, secrecy::SecretVec, AuthUser, UserStore,
+};
+use hyper::StatusCode;
+use serde::Deserialize;
+use std::{collections::HashMap, fmt::Display, hash::Hash, sync::Arc};
+use thiserror::Error;
+use tokio::sync::RwLock;
+
+pub type SocketBobMemoryStore = MemoryStore<Hostname, BobClient>;
+pub type BobStore<Id> = HashMap<Id, BobClient>;
+pub type SharedBobStore<Id> = Arc<RwLock<BobStore<Id>>>;
+
+#[derive(Debug, Error)]
+pub enum AuthError {
+    #[error("couldn't load user")]
+    LoadError,
+    #[error("couldn't lock user store")]
+    PoisonError,
+}
+
+/// Optional credentials for a BOB cluster
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Default, Deserialize)]
+pub struct Credentials {
+    pub login: String,
+    pub password: String,
+}
+
+/// Data needed to login to a BOB cluster
+impl AuthUser<Hostname> for BobClient {
+    fn get_id(&self) -> Hostname {
```

Comment:
This is the `id` of user. Thus `Hostname` is not enough here. There can be multiple authorizations to the same node, but with different users, so I suggest to use a separate type, that will be the combination of `Hostname` and user name